src/ciadmin/generate/worker_pools.py

```diff
@@ -221,10 +225,18 @@ def get_azure_provider_config(
                 f"{subscription_id}/resourceGroups/{rgroup}/providers/"
                 f"Microsoft.Network/virtualNetworks/{vnet}/subnets/{snet}"
             )
-            imageReference_id = (
-                f"{subscription_id}/resourceGroups/{image_rgroup}/providers/"
-                f"Microsoft.Compute/images/{ImageId}-{DeploymentId}"
-            )
+            if version is not None:
+                ImageId = image.image_id(provider_id, "name")
+                imageReference_id = (
+                    f"{subscription_id}/resourceGroups/rg-packer-worker-images/providers/"
```

Comment:
What is `rg-packer-worker-images`? Does it need to be hardcoded here? At the very least this should probably be raised to a constant at the top of the file rather than buried here. Preferably, `image_rgroup` would be correct both with and without a `version`. It looks like that's already in the yml, even?